### PR TITLE
ICub effort bridge

### PR DESCRIPTION
**Pull request by MartaLorenzini**

### CODE REVIEW

File: human-viz-bridge/src/iCubEffortBridge.cpp [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+#include <yarp/os/Publisher.h>
+#include <yarp/os/RFModule.h>
+#include <yarp/os/Time.h>
+#include <yarp/sig/Vector.h>
+
+#include <iDynTree/Core/EigenHelpers.h>
+
+#include <algorithm>
+#include <iostream>
+#include <limits.h>
```
Comment (francesco-romano) [inline human-viz-bridge/src/iCubEffortBridge.cpp]: use the c++ style header, i.e. `climits`

File: human-viz-bridge/src/iCubEffortBridge.cpp [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+    return ret;
+}
+
+class iCubEffortPublisherModule : public RFModule 
+{
+    iDynTree::VectorDynSize robotTorques;
+    PolyDriver m_robot;
+    ITorqueControl *robotITorqueControl;
+    map<std::string, EffortPublisher> effortMap;
+    int period;
```
Comment (francesco-romano) [inline human-viz-bridge/src/iCubEffortBridge.cpp]: this is an integer, but the `RFModule` wants a double

File: human-viz-bridge/src/iCubEffortBridge.cpp [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+
+    ret.sec  = sec_part;
+    ret.nsec = nsec_part;
+    return ret;
+}
+
+class iCubEffortPublisherModule : public RFModule 
+{
+    iDynTree::VectorDynSize robotTorques;
+    PolyDriver m_robot;
```
Comment (francesco-romano) [inline human-viz-bridge/src/iCubEffortBridge.cpp]: you should be consistent in how you declare members.
I prefer the `m_` version anyway.

File: human-viz-bridge/src/iCubEffortBridge.cpp [hunk 4]
```diff
@@ -0,0 +1,330 @@
...
+            sensor_msgs_Temperature &effortMsg = effort.publisher->prepare();
+            effort.effortMsg.header.stamp = currentTime;
+            
+            double effort_temp = 0;
+            for (size_t& effortIdx : effort.jointDoFs) {
+                effort_temp += std::abs(robotTorques.getVal(effortIdx));
+            }
+            effort.effortMsg.temperature = effort_temp;
+            
+            //cout << effortList.first << " in the reference frame: " << effort.effortMsg.header.frame_id << " with variance: " << effortList.second.effortMsg.variance << " and torque: " << effort.effortMsg.temperature << endl;
```
Comment (francesco-romano) [inline human-viz-bridge/src/iCubEffortBridge.cpp]: remove the unnecessary comments in the final version

File: human-viz-bridge/src/iCubEffortBridge.cpp [hunk 5]
```diff
@@ -0,0 +1,330 @@
...
+//---------------------------------------------------------------------------
+//Utility functions for parsing INI file
+static bool parseStringListOption(const Value &option, vector<string> &parsedList);
+//---------------------------------------------------------------------------
+
+struct EffortPublisher {
+    string linkEffort;
+    vector<size_t> jointDoFs;
+    Publisher<sensor_msgs_Temperature> *publisher;
+    sensor_msgs_Temperature effortMsg;
```
Comment (francesco-romano) [inline human-viz-bridge/src/iCubEffortBridge.cpp]: can we try to remove this variable and use the one returned by `prepare()`

File: human-viz-bridge/src/iCubEffortBridge.cpp [hunk 6]
```diff
@@ -0,0 +1,330 @@
...
+            }
+            if (!effortList.second.publisher->topic(topicName)) {
+                yError() << "Failed to create publisher to" << effortList.second.linkEffort;
+                return false;
+            }
+            
+        }
+        
+        for (auto& effortList : effortMap) {
+            string linkEffort;
```
Comment (francesco-romano) [inline human-viz-bridge/src/iCubEffortBridge.cpp]: You can probably move these lines inside the previous `for` given it iterates over the same elements

File: human-viz-bridge/src/iCubEffortBridge.cpp [hunk 7]
```diff
@@ -0,0 +1,330 @@
...
+            EffortPublisher &effort = effortList.second;
+            if (effort.publisher) {        
+                effort.publisher->interrupt();
+                effort.publisher->close();
+                delete effort.publisher;
+                effort.publisher = 0;
+            }
+        }
+        effortMap.clear();
+
```
Comment (francesco-romano) [inline human-viz-bridge/src/iCubEffortBridge.cpp]: zero also the torque control interface

File: human-viz-bridge/src/iCubEffortBridge.cpp (hunk 5 above)
Comment (MartaLorenzini) [inline human-viz-bridge/src/iCubEffortBridge.cpp]: If we want to do that we have to put all the assignments for that variable in the update section

**Comment by Yeshasvitvs** [comment]
@diegoferigo Can we close this PR?

**Comment by diegoferigo** [comment]
@Yeshasvitvs Yes this is very old and the effort bridge has been rewritten at least twice in the meantime. Closing this PR.

**Pull request closed without merging**